packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -1,78 +1,44 @@
-                        <div className='settings-language'>
-                            <DesktopWrapper>
-                                <FormSubHeader title={localize('Select Language')} />
-                            </DesktopWrapper>
-                            <div className='settings-language__language-container'>
-                                {allowed_language_keys.map(language_key => {
-                                    return (
-                                        <LanguageRadioButton
-                                            key={language_key}
-                                            id={language_key}
-                                            language_code={language_key}
-                                            is_current_language={values.language_code === language_key}
-                                            name='language-radio-group'
-                                            onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-                                                setFieldValue('language_code', event.target.value)
-                                            }
-                                        />
-                                    );
-                                })}
-                            </div>
-                        </div>
-                        <FormFooter>
-                            <Button
-                                className='account-form__footer-btn'
-                                type='submit'
-                                data-testid={'submit-button'}
-                                has_effect
-                                text={localize('Submit')}
-                                large
-                                primary
-                                is_disabled={isCurrentLanguage(values.language_code)}
-                            />
-                        </FormFooter>
-                    </form>
-                );
-            }}
-        </Formik>
+        <div className='settings-language'>
+            <FormSubHeader title={localize('Select Language')} />
+            <div className='settings-language__language-container'>
+                {allowed_language_keys.map(language_key => {
+                    return (
+                        <LanguageRadioButton
+                            key={language_key}
+                            id={language_key}
+                            language_code={language_key}
+                            is_current_language={current_language === language_key}
+                            name='language-radio-group'
+                            onChange={async () => {
+                                changeSelectedLanguage(language_key);
+                                await i18n.changeLanguage?.(language_key);
+                            }}
+                        />
+                    );
+                })}
+            </div>
+        </div>
```

Comment:
how are we closing this language pop up once user selects a language?

Reply:
I think he's relying on the close button on the top right corner instead of closing the window after language change.

Seems like an ok solution to me.

Reply:
yes. According to design description, user should stay on this page after changing the language.
and they can still click 'X' and leave account settings